runtime/onert/backend/train/TensorManager.h

```diff
@@ -32,6 +32,13 @@ namespace train
 
 class TensorManager
 {
+public:
+  // Minimum alignment for buffers
+  //
+  // _align is equal to EIGEN_MAX_ALIGN_BYTES, though including Eigen headers
+  // here to get that symbol may not be a good idea.
+  static constexpr uint64_t _align = 16;
```

Comment:
_Inline variables_ are supported since c++17, but onert is still using c++14. So you have to define this variables in the cpp file(`TensorManager.cpp`) to prevent multiple definition error.

Reply:
I think it is okay. There are several places in use in our code. `_align` is not odr-used.

Reply:
@ragmani 

> _Inline variables_ are supported since c++17, but onert is still using c++14. So you have to define this variables in the cpp file(`TensorManager.cpp`) to prevent multiple definition error.

This is a class member variable. Should I need to consider multiple definition error in this case?

Reply:
`static constexpr` means `static constexpr inline` implicitly since c++17.  _inline keyword_ prevents multiple definition by header files that has definitions when linking object files with the header files. So defining `static constexpr` member in a header file is OK in c++17. However, c++14 does not introduce _inline variables_ that can allow using _inline keywords_ to variables such as static member.
If `TensorManager` is generated in multiple object files and the object are linked, c++ compiler will emit multiple definition error.

Reply:
As I mentioned in https://github.com/Samsung/ONE/pull/12043#discussion_r1393623601, in this PR's case, it looks okay since there is no odr-use. See https://en.cppreference.com/w/cpp/language/definition#ODR-use. You can find several occurrences the same usage by search "static constexpr" in our code base.

Reply:
I guess there are two ways.
1. Fix it when multiple definition error occors because this code willl not have this problem if onert starts using c++17.
2. Fix it now.

I think No.1 way is not bad.

Reply:
@ragmani As I told you offline, `cker/Shape.h` has:

```cc
class Shape
{
  static constexpr int kMaxSmallSize = 5;
```

`cker/Shape.h` is included in almost all `cker/operation/*.h`.
`cker/operation/*.h` is included by the corredponding `*Layer.cc`.

I think the address of `kMaxSmallSize` is not required at all.
Compiler is smart to embed the value w/o preparing memory storage for `kMaxSmallSize`.

Please find 

- my 1st link 

<sub> https://en.cppreference.com/w/cpp/language/definition#ODR-use </sub>
```
ODR-use

Informally,
- an object is odr-used if its value is read (unless it is a compile time constant) or written, its address is taken, or a reference is bound to it, 
```

NOTE that _unless it is a compile time constant_.

- and several SO answers (e.g. https://stackoverflow.com/questions/23491781/when-is-a-variable-odr-used-in-c14)

```
Informally, odr-use of a variable means the following:

If any expression anywhere in the program takes the address of or binds a reference directly to an object, this object must be defined.
```

Reply:
I tried an experiment about it. If `const` or `constexpr` is used, neither "static members" nor "namespace variables" had the problem I mentioned. I don't know the exact reason why `const` or `constexpr` allows "static members" and "namespace variables" to be multi-defined.
Anyway, there is no problem with current code. Sorry for the delay due to incorrect knowledge.